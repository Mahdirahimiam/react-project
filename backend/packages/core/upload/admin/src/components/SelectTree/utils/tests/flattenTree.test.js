import flattenTree from '../flattenTree';

const FIXTURE = [
  {
    value: 'f-1',
    label: 'Folder 1',
  },

  {
    value: 'f-2',
    label: 'Folder 2',
    children: [
      {
        value: 'f-2-1',
        label: 'Folder 2-1',
      },

      {
        value: 'f-2-2',
        label: 'Folder 2-2',
        children: [
          {
            value: 'f-2-2-1',
            label: 'Folder 2-2-1',
          },
        ],
      },
    ],
  },
];

describe('flattenTree', () => {
  test('flattens the passed tree structure properly', () => {
    expect(flattenTree(FIXTURE)).toMatchSnapshot();
  });
});
