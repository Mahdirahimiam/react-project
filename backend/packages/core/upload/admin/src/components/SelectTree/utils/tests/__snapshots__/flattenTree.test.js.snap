// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flattenTree flattens the passed tree structure properly 1`] = `
[
  {
    "depth": 0,
    "label": "Folder 1",
    "parent": undefined,
    "value": "f-1",
  },
  {
    "children": [
      {
        "label": "Folder 2-1",
        "value": "f-2-1",
      },
      {
        "children": [
          {
            "label": "Folder 2-2-1",
            "value": "f-2-2-1",
          },
        ],
        "label": "Folder 2-2",
        "value": "f-2-2",
      },
    ],
    "depth": 0,
    "label": "Folder 2",
    "parent": undefined,
    "value": "f-2",
  },
  {
    "depth": 1,
    "label": "Folder 2-1",
    "parent": "f-2",
    "value": "f-2-1",
  },
  {
    "children": [
      {
        "label": "Folder 2-2-1",
        "value": "f-2-2-1",
      },
    ],
    "depth": 1,
    "label": "Folder 2-2",
    "parent": "f-2",
    "value": "f-2-2",
  },
  {
    "depth": 2,
    "label": "Folder 2-2-1",
    "parent": "f-2-2",
    "value": "f-2-2-1",
  },
]
`;
